### PR TITLE
Support to build with Docker container

### PR DESCRIPTION
**Pull request by flier**
Support to build with Docker container

### CODE REVIEW

**Comment by flier** [comment]
replace PR #9 Add a Dockerfile for project

**Comment by codefromthecrypt** [comment]
Maybe add a readme what this is used by or what it is for?

File: ci/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Developer use of CI Docker images
+
+Two flavors of Envoy Docker images, based on Ubuntu and CentOS Linux, are built.
```
Comment (codefromthecrypt) [inline ci/README.md]: did you mean to say envoy here?

File: ci/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+## Building and running tests as a developer
+
+## Testing changes to the build image as a developer
+
+While all changes to the build image should eventually be upstreamed, it can be useful to test those changes locally before sending out a pull request. To experiment with a local clone of the upstream build image you can make changes to files such as build_image.sh locally and then run:
+
+```sh
+$ cd ci
+$ ./build-image.sh  # Wait patiently for quite some time
```
Comment (codefromthecrypt) [inline ci/README.md]: does this emit something to console when complete? Does a built binary exist at a known path on the image when complete? Answers to these questions could be in this README

File: ci/README.md (hunk 1 above)
Comment (flier) [reply ci/README.md]: Oops, just copy the text form Envoy :)

**Comment by codefromthecrypt** [comment]
thanks, so I can use this image to test code changes and/or build a binary that can be found at the path dist/zipkin-cpp-0.3.1-Linux-x86_64.tar.gz once the build is done. nice!

**Comment by flier** [comment]
Sure, but it seems you need use `git:develop` branch instead of outdated 0.3.1

**Comment by codefromthecrypt** [comment]
I will try this locally

**Comment by codefromthecrypt** [comment]
wondering if the build also makes the examples? If so, it could be worth mentioning as people could try them (possibly by running the resulting container!)

**Comment by codefromthecrypt** [comment]
slight glitch running locally as it fails a unit test
```
$ ./build-dist.sh git:develop
--snip--
QueueImpl.cpp:(.text._ZN7RdKafka9QueueImplD1Ev[_ZN7RdKafka9QueueImplD1Ev]+0x39): undefined reference to `rd_kafka_queue_destroy'
collect2: error: ld returned 1 exit status
make[2]: *** [test/unittest] Error 1
make[1]: *** [test/CMakeFiles/unittest.dir/all] Error 2
make: *** [all] Error 2
test/CMakeFiles/unittest.dir/build.make:190: recipe for target 'test/unittest' failed
CMakeFiles/Makefile2:1091: recipe for target 'test/CMakeFiles/unittest.dir/all' failed
Makefile:140: recipe for target 'all' failed
```

**Comment by flier** [comment]
> QueueImpl.cpp:(.text._ZN7RdKafka9QueueImplD1Ev[_ZN7RdKafka9QueueImplD1Ev]+0x39): undefined reference to `rd_kafka_queue_destroy'

Seems caused by missing librdkafka C library, it should be fixed in `feature/docker-build` branch

You may try to use the `feature/docker-build` branch
```sh
$ ./build-dist.sh git:feature/docker-build
```

**Comment by flier** [comment]
In fact, the build script will map the `dist` folder to the container, you could check it for your package :)

**Pull request merged**